identity/src/main/java/life/qbic/identity/domain/model/token/PersonalAccessTokenEncoder.java

```diff
@@ -0,0 +1,172 @@
+import javax.crypto.spec.SecretKeySpec;
+import life.qbic.logging.api.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+/**
+ * Encodes personal access tokens
+ */
+@Service
+public class PersonalAccessTokenEncoder implements TokenEncoder {
+
+  private static final Logger log = logger(PersonalAccessTokenEncoder.class);
+
+  private final String salt;
+  private final int iterationCount;
+
+  private static final int EXPECTED_MIN_ITERATION_COUNT = 10_000;
+
+  private static final int ITERATION_COUNT_INDEX = 0; // the index of the iteration count in the encoded token
+  private static final int SALT_INDEX = 1; // the index of the salt content in the encoded token
+  private static final int HASH_INDEX = 2; // the index of the hash content in the encoded token
+
+  public PersonalAccessTokenEncoder(
+      @Value("${qbic.access-token.salt}") String salt,
+      @Value("${qbic.access-token.iteration-count}") int iterationCount) {
+    this.salt = requireNonNull(salt, "salt must not be null");
+    if (this.salt.isBlank()) {
+      throw new IllegalArgumentException("salt must not be blank");
+    }
+    if (iterationCount <= 0) {
+      throw new IllegalArgumentException("iterationCount must be greater than 0");
+    }
+    if (iterationCount <= EXPECTED_MIN_ITERATION_COUNT) {
+      log.warn(
+          "Low iteration count for token encryption detected n=%d. Expected minimal iteration count is n=%d".formatted(
+              iterationCount, EXPECTED_MIN_ITERATION_COUNT));
+    }
+    this.iterationCount = iterationCount;
+  }
+
+  private record EncryptionSettings(String cipher, int keyBitSize) {
+
+  }
+
+  private static final EncryptionSettings ENCRYPTION_SETTINGS = new EncryptionSettings("AES", 256);
+
+
+  @Override
+  public String encode(char[] token) {
+    String saltyString = requireNonNull(this.salt, "this.salt must not be null");
+    if (saltyString.isBlank()) {
+      throw new RuntimeException("Salt cannot be blank");
+    }
+    var iterationCountCopy = this.iterationCount;
+    if (iterationCountCopy <= EXPECTED_MIN_ITERATION_COUNT) {
+      log.warn("Low iteration count for personal access token encryption");
+    }
+    byte[] saltBytes = saltyString.getBytes(StandardCharsets.UTF_8);
+    byte[] hash = pbe(token, saltBytes, iterationCountCopy);
+    return iterationCountCopy + ":" + toHex(saltBytes) + ":" + toHex(hash);
+  }
+
+  @Override
+  public boolean matches(char[] token, String encodedToken) {
+    byte[] readSalt = readSalt(encodedToken);
+    if (!Arrays.equals(readSalt, this.salt.getBytes(StandardCharsets.UTF_8))) {
+      log.warn("Personal access token has different salt than currently configured.");
+    }
+    int readIterationCount = readIterationCount(encodedToken);
+    if (readIterationCount != this.iterationCount) {
+      log.warn("Personal access token has different iterations than currently configured.");
+    }
```

Comment:
This is not an issue, since we store the iteration count with the encrypted token